packages/server/src/feature/teacher/Repository.ts

```diff
@@ -0,0 +1,190 @@
+    }, query);
+
+    if (params.limit !== undefined) {
+      query.limit(params.limit);
+    }
+
+    if (params.offset !== undefined) {
+      query.offset(params.offset);
+    }
+
+    const result = await query;
+    return result.map(teacher => this.transformTeacher(teacher));
+  }
+
+  async getTeachersCount(filter: TeacherFilter): Promise<number> {
+    const query = this._db
+      .select({ count: sql<number>`count(*)` })
+      .from(userTable);
+
+    this.applyFilter(filter, query);
+
+    const result = await query;
+    return Number(result[0].count);
+  }
+
+  async getTeacher(id: string): Promise<Teacher | null> {
+    const result = await this._db
+      .select()
+      .from(userTable)
+      .where(and(eq(userTable.id, id), eq(userTable.role, "teacher")))
+      .limit(1);
+
+    if (result.length === 0) {
+      return null;
+    }
+
+    return this.transformTeacher(result[0]);
+  }
+
+  async putTeacher(id: string, teacherUpdateParams: PutTeacherParams, hashedPassword: string): Promise<Teacher> {
+    try {
+      const dateNow = new Date();
+
+      const updatedTeacher = await this._db
+        .update(userTable)
+        .set({
+          email: teacherUpdateParams.email,
+          password: hashedPassword,
+          first_name: teacherUpdateParams.firstName,
+          last_name: teacherUpdateParams.lastName,
+          created_at: dateNow,
+          updated_at: dateNow,
+        })
+        .where(and(eq(userTable.id, id), eq(userTable.role, "teacher")))
+        .returning();
+
+      if (updatedTeacher.length === 0) {
+        throw TeacherError.updateFailed(`Failed to update teacher with ID "${id}".`);
+      }
+      return this.transformTeacher(updatedTeacher[0]);
+    } catch (error: any) {
+      if (error.cause.code === "23505") {
+        throw TeacherError.alreadyExists(
+          `Teacher with email "${teacherUpdateParams.email}" already exists.`,
+        );
+      }
+      throw TeacherError.updateFailed(
+        "Unexpected error during teacher update",
+        error,
+      );
+    }
+  }
+
+  async patchTeacher(id: string, teacherUpdateParams: PatchTeacherParams): Promise<Teacher> {
+    try {
+      const updateData: Partial<typeof userTable.$inferInsert> = {};
+
+      if (teacherUpdateParams.email !== undefined) {
+        updateData.email = teacherUpdateParams.email;
+      }
+      if (teacherUpdateParams.password !== undefined) {
+        updateData.password = teacherUpdateParams.password;
+      }
+      if (teacherUpdateParams.firstName !== undefined) {
+        updateData.first_name = teacherUpdateParams.firstName;
+      }
+      if (teacherUpdateParams.lastName !== undefined) {
+        updateData.last_name = teacherUpdateParams.lastName;
+      }
+
+      updateData.updated_at = new Date();
+
+      const result = await this._db
+        .update(userTable)
+        .set(updateData)
+        .where(and(eq(userTable.id, id), eq(userTable.role, "teacher")))
+        .returning();
```

Comment:
éviter de return le mdp (même hashé)
```suggestion
      .returning({
        id: userTable.id,
        email: userTable.email,
        firstName: userTable.first_name,
        lastName: userTable.last_name,
        updatedAt: userTable.updated_at,
```
      });

Reply:
j'ai le transformTeacher qui s'en occupe mais ok :)